feat: add preview for Microsoft Word documents

Alchez opened the pull request:
Libraries added (to keep consistent with Frappe Drive):
- https://github.com/VolodymyrBaydalka/docxjs
- https://github.com/Stuk/jszip (un-packaged peer dependency for docxjs)

--

Sample File:

[de876b68_TESTING.docx](https://github.com/agritheory/cloud_storage/files/13792350/de876b68_TESTING.docx)

Preview:

![localhost_8004_app_file_92d040d462](https://github.com/agritheory/cloud_storage/assets/13396535/4dbe506c-52e6-4563-9ad4-a311d9c3a9af)

Commit:
feat: add preview for Microsoft Word documents

Alchez commented on cloud_storage/public/js/cloud_storage.bundle.js:
```diff
@@ -1 +1,3 @@
 import './overrides'
+
+frappe.Docx = require('docx-preview')
```
This seems to be required, otherwise the browser throws a "Cannot use import statement outside a module" error in the console.

Alchez committed:
fix: fallback to file name if file type isn't found

Commit:
fix: extension string check

agritheory commented:
@Alchez Notes from our call:
 - Rebase this to a new V15 branch
 - Resolve the error that came up with an http linked docx file

bhattdevarsh committed:
fix: encoded url on get_content (#51)

agritheory commented:
@Alchez Please update documentation to include this feature

Commit:
Merge remote-tracking branch 'upstream/version-15' into feat-docx-preview

Commit:
fix: allow http link files

Commit:
docs: add documentation for Microsoft Word file preview

Commit:
test: update moto dependencies

Alchez commented:
@agritheory I've fixed up the remaining issues and added some basic documentation for Word files. The Frappe tests in the GHA are failing due to typing validations on test objects, but they're running correctly on my local system. I think this one is good to be reviewed again.

HKuz committed:
test: fix delete file test (#63)

* ci: add frappe black to CI (#57)

* ci: add frappe black to CI

* chore: black

* ci: fix v15 tests

* ci: run redis server

---------

Co-authored-by: Tyler Matteson <support@agritheory.dev>
Co-authored-by: Heather Kusmierz <heather@agritheory.dev>
Co-authored-by: Rohan Bansal <rohan@agritheory.dev>

agritheory committed:
chore: remove ruff

agritheory committed:
Merge branch 'version-15' into feat-docx-preview

agritheory committed:
fix: remove broken isort lint, black

agritheory committed:
Merge branch 'version-15' of github.com:agritheory/cloud_storage into feat-docx-preview

agritheory commented:
@Alchez Can you give this another once-over, I think it's ready to go.

Commit:
fix: black format and isort

Commit:
Merge remote-tracking branch 'upstream/version-15' into feat-docx-preview

Commit:
chore: update poetry.lock

Alchez commented:
@agritheory I think this one is good to go, I've added a fix for the poetry lockfile
